tools/sense_studio/sense_studio.py

```diff
@@ -337,6 +338,40 @@ def show_video_list(split, label, path):
     return render_template('video_list.html', folders=folder_id, split=split, label=label, path=path)
 
 
+@app.route('/record_video/<split>/<label>/<path:path>')
+def record_video(split, label, path):
+    """
+    Show the list of videos for the given split, class label and project.
+    If the necessary files for annotation haven't been prepared yet, this is done now.
+    """
+    path = f'/{urllib.parse.unquote(path)}'  # Make path absolute
+    split = urllib.parse.unquote((split))
+    label = urllib.parse.unquote(label)
+    return render_template('record_video.html', split=split, label=label, path=path)
+
+
+@app.route('/video_saving/<split>/<label>/<path:path>', methods=['POST'])
+def save_video(path, label, split):
+    path = f'/{urllib.parse.unquote(path)}'  # Make path absolute
+    if request.method == 'POST':
+        file = request.files['video']
+        temp_file = os.path.join(path, "temp_video.webm")
+        with open(temp_file, "wb") as vid:
+            video_stream = file.read()
+            vid.write(video_stream)
+        output_path = os.path.join(path, "videos_" + split, label)
+
+        os.path.join(output_path, f"video_0.mp4")
+        # find a video name that is not used yet
+        for i in range(100000):
+            output_file = os.path.join(output_path, f"video_{i}.mp4")
+            if not os.path.isfile(output_file):
+                break
```

Comment:
This could maybe be done more efficiently with a `glob` (not requiring that many `isfile` checks)